chainer/function_node.py

```diff
@@ -1009,6 +1009,44 @@ def grad(outputs, inputs, grad_outputs=None, grad_inputs=None, set_grad=False,
                 'len(inputs) = {}, len(grad_inputs) = {}'
                 .format(len(inputs), len(grad_inputs)))
 
+    # Check if all the inputs are chainerx arrays and if so
+    # Relies in chainerx.grad function
+    chx_inputs = sum(map(lambda x: x._has_chainerx_array
```

Comment:
How about this?
`sum([False if x is None else x._has_chainerx_array for x in inputs])`